bin/docker-push

```diff
@@ -41,8 +57,9 @@ if [ "$DOCKER_BUILD_CACHE_FROM" == "available" ]; then
   done <<< "$(grep -i '^FROM.* AS ' ${DOCKERFILE} | awk '{print $4}')"
 fi
 
-echo "Pushing ${DOCKERTAG}:latest as ${CI_SHA1}, ${CI_REF}, ${CI_BUILD_NUM} to ${EXTERNAL_REGISTRY_BASE_DOMAIN}/${REPOSITORY_NAME}"
-
-tag_push "${DOCKERTAG}:latest" "${EXTERNAL_REGISTRY_BASE_DOMAIN}/${REPOSITORY_NAME}:${CI_SHA1}"
-tag_push "${DOCKERTAG}:latest" "${EXTERNAL_REGISTRY_BASE_DOMAIN}/${REPOSITORY_NAME}:${CI_REF}"
-tag_push "${DOCKERTAG}:latest" "${EXTERNAL_REGISTRY_BASE_DOMAIN}/${REPOSITORY_NAME}:build_${CI_BUILD_NUM}"
+for DOCKER_TAG_VERSION in ${ALL_DOCKER_TAG_VERSIONS[@]}; do
+  if [ "${DOCKER_LATEST_TAG}" != "${DOCKER_TAG_VERSION}" ]; then
+    echo "Pushing ${DOCKERTAG}:${DOCKER_LATEST_TAG} as ${EXTERNAL_REGISTRY_BASE_DOMAIN}/${REPOSITORY_NAME}:${DOCKER_TAG_VERSION}"
+    tag_push "${DOCKERTAG}:${DOCKER_LATEST_TAG}" "${EXTERNAL_REGISTRY_BASE_DOMAIN}/${REPOSITORY_NAME}:${DOCKER_TAG_VERSION}"
```

Comment:
This looks great! I am looking at where else `:latest` might be inferred and I think this might have `./bin/docker-build` implications. After crawling through this a bit I see that the `$DOCKERTAG` variable is not such a great name since it looks to be the image _and_ (potentially) tag name!

If someone has defined the tag name in the `$DOCKERTAG` variable then we'll hit conditions where this line could read `image-name:sometag:$DOCKER_LATEST_TAG` (which would break).

The root of the problem here is that `$DOCKERTAG` isn't named very well at all. To accommodate this how would you feel about adding in a check to assure we don't have a `:` in the `$DOCKERTAG` variable.

```bash
# Adding override convention of DOCKERTAG
DOCKER_LATEST_TAG=${DOCKER_LATEST_TAG:-latest}
if [[ "${DOCKERTAG}" =~ ":" ]]; then
  echo "The \$DOCKERTAG (${DOCKERTAG}) var contains a tag name. Skipping adding \$DOCKER_LATEST_TAG (${DOCKER_LATEST_TAG})."
else
  echo "The \$DOCKERTAG (${DOCKERTAG}) var does not contain a default tag. Appending \$DOCKER_LATEST_TAG ($DOCKER_LATEST_TAG)."
  DOCKERTAG="${DOCKERTAG}:${DOCKER_LATEST_TAG}"
fi
```
(Feel free to trash the above block, I just want to make sure we're not breaking implementations where people are using a tag in `$DOCKERTAG` field.)

I could be mistaken too - I'm jumping into this part of `rok8s-scripts` and it's a little hard to tell where some of these variables are used :-\ -- thanks again for your patience and contribution.

Reply:
I am also asking internally to see if anyone who originally wrote this piece has some insight into intent of `DOCKERTAG`

Reply:
Ok - after tracing through the history of this repo, it would appear that `docker-push` has always used the `docker tag ${DOCKERTAG}:latest` convention. This means no one should have ever been able to set `DOCKERTAG=image:tag` and have `./bin/docker-push` work properly. This leads me to believe all implementations that currently exist will _not_ have a tag name in `DOCKERTAG` variable.

Since the `docker-build` script needs `DOCKERTAG` to do its builds, we should at least add the same block to set the `DOCKER_LATEST_TAG` in `./bin/docker-build`. I also see `:latest` used in `./bin/docker-save`, so this should also be replaced with the `DOCKER_LATEST_TAG` convention.

So, long story short: We should be fine to approve & merge as long as `./bin/docker-build` and `./bin/docker-save` are updated to utilize the new `DOCKER_LATEST_TAG` override tag.

Sound ok?

Reply:
Done.  See https://github.com/reactiveops/rok8s-scripts/pull/213/commits/8db98c995a77875a6ef5c3907f20316cc62ee407